2_process/src/munge_GFv1_catchments.R

```diff
@@ -0,0 +1,146 @@
+  #' the catchment boundaries are adjusted for NHGFv1 segments that were split in the 
+  #' `delaware-model-prep` pipeline (https://github.com/USGS-R/delaware-model-prep) 
+  #' 
+  #' @param prms_lines sf object containing the PRMS river segments for the DRB
+  #' prms_lines must contain variables subsegid and geometry
+  #' @param segs_w_comids data frame containing the PRMS segment ids and the contributing COMID's
+  #' segs_w_comids must contain variables PRMS_segid and COMID
+  #' @param prms_hrus sf (multi)polygon containing the HRU's from the NHGFv1
+  #' @param segs_split character vector indicating the PRMS_segid of the segments that were split in 
+  #' the delaware-model-prep pipeline
+  #' @param crs_out numeric EPSG code indicating desired crs. Defaults to EPSG: 4326, WGS84.
+  #
+  
+  # 1. Fill PRMS lines data frame so that every segment has a value for the attribute
+  # `hru_segment`, which can be used as a key to join each segment to its corresponding HRU polygons.
+  prms_hrus_filled <- prms_lines %>%
+    # Transform prms_lines into a list with a data frame for each PRMS segment (represented by a unique subsegid) 
+    split(.,.$subsegid) %>%
+    # For each PRMS segment, find the HRU with the greatest overlap and add as an attribute
+    lapply(.,function(x){
+      x$hru_segment <- ifelse(x$subsegseg %in% prms_hrus$hru_segment,
+                              x$subsegseg,
+                              find_intersecting_hru(x,prms_hrus))
+      return(x)
+    }) %>%
+    # Reformat list into a data frame with one row per PRMS segment
+    bind_rows()
+  
+  # 2. For select segments, redefine the catchment polygon using the contributing NHDv2 catchments
+  prms_splitsegs <- prms_hrus_filled %>%
+    # Filter PRMS segments to only include select segments that require special handling
+    filter(subsegid %in% segs_split) %>%
+    # Transform PRMS segments into a list with a data frame for each segment
+    split(.,.$subsegid) %>%
+    # For each segment requiring special handling, return the polygon that represents the catchment area
+    lapply(.,return_nhdv2_cat_shps, segs_w_comids = segs_w_comids) %>%
+    # Reformat list into a data frame with one row per PRMS segment
+    bind_rows()
+  
+  # 3. Concatenate munged catchment polygons
+  catchments_ls <- list()
+  
+  # Fill empty list with appropriate catchments
+  for(i in seq_along(prms_hrus_filled$subsegid)){
+    
+    prms_line <- prms_hrus_filled[i,]
+    
+    # If PRMS segment is one of the segments that require special handling, 
+    # retrieve NHDPlusV2 catchments and use those as the catchment boundary.
+    # Else, represent the catchment boundaries using intersecting HRU polygon.
+    if(prms_line$subsegid %in% segs_split){
+      cat <- prms_splitsegs %>%
+        filter(PRMS_segid == prms_line$subsegid) %>%
+        mutate(hru_segment = prms_line$subsegid,
+               hru_id = NA) %>%
+        select(PRMS_segid,hru_segment,hru_id,geometry) %>%
```

Comment:
`hru_area` is important to keep here too because it allows us to compute the area-weighted calculations of across the different data points.

Reply:
This particular code chunk refers to the NHDv2-derived "new" HRU's so there isn't an attribute called `hru_area` to select. So I used `sf::st_area` to calculate the area of the new HRU(s) and named that column `hru_area_m2`.